lib/node_modules/@stdlib/array/bool/test/test.reduce.js

```diff
@@ -0,0 +1,218 @@
+var isFunction = require( '@stdlib/assert/is-function' );
+var Boolean = require('@stdlib/boolean/ctor');
+var BooleanArray = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof BooleanArray, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `reduce` method', function test( t ) {
+	t.strictEqual( hasOwnProp( BooleanArray.prototype, 'reduce' ), true, 'has property' );
+	t.strictEqual( isFunction( BooleanArray.prototype.reduce ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new BooleanArray( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.map.reduce( value, reducer );
+		};
+	}
+
+	function reducer( acc, value ) {
+		if ( value ) {
+			return acc + 1;
+		}
+		return acc;
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new BooleanArray( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.reduce( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if not provided an initial value when operating on an empty complex number array', function test( t ) {
+	var arr;
+
+	arr = new BooleanArray( 0 );
+	t.throws( foo, Error, 'throws an error' );
+	t.end();
+
+	function foo() {
+		arr.reduce( reducer );
+	}
+
+	function reducer( acc, value ) {
+		if ( value ) {
+			return acc + 1;
+		}
+		return acc;
+	}
+});
+
+tape( 'the method uses the first element of the array as the initial value when an initial value is not provided', function test( t ) {
+	var valueArray;
+	var accArray;
+	var expected;
+	var actual;
+	var arr;
+	var ind;
+
+	arr = new BooleanArray( [ true, false, true, false ] );
+	accArray = [ true, false, false ];
+	valueArray = [ false, true, false ];
+	expected = false;
+	actual = arr.reduce( reducer );
+
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	t.end();
+
+	function reducer( acc, value, index ) {
+		ind = index-1;
+		t.strictEqual( acc, accArray[ ind ], 'returns expected value' );
+		t.strictEqual( value, valueArray[ ind ], 'returns expected value' );
+		return Boolean( acc & value );
```

Comment:
`&` v `&&`

Reply:
You also don't need the `Boolean` constructor here for what you are wanting to do.